### PR TITLE
【BIT】引擎能力补齐：paddle.nn.functional.[cross_entropy ctc_loss dice_loss] paddle.geometric.[sample_neighbors send_*]

### PR DESCRIPTION
**Pull request by cangtianhuang**
### 1. paddle.nn.functional.cross_entropy

cross_entropy 的 `label` 参数需根据 *输入的类别数*、`soft_label`、`label_smoothing` 等参数确定生成值，分为独热标签或整数标签两类。为其撰写了相应的生成方法，并在 get_numpy_tensor 方法中增加分支处理

### 2. paddle.nn.functional.ctc_loss

ctc_loss 的 `label` 参数表示经过 padding 的标签序列、`input_lengths` 表示输入 log_probs 数据中每个序列的长度， `label_lengths` 表示 label 中每个序列的长度。根据 api 要求为其撰写了生成方法，并在 get_numpy_tensor 方法中增加分支处理

### 3. paddle.nn.functional.dice_loss

dice_loss 的 `label` 参数表示正确的标注数据，仅需生成类别数范围内的随机数。为其撰写了生成方法，并在 get_numpy_tensor 方法中增加分支处理

### 4. paddle.geometric.sample_neighbors

在 https://github.com/PFCCLab/PaddleAPITest/pull/23 中直接通过测试，但是在其他环境中报错。因此根据文档，为其输入图的 CSC（Compressed Sparse Column，压缩列信息）生成符合要求的参数，包括 `row`、`colptr`、`input_nodes`、`eids`、`perm_buffer` 等，具体生成方法见代码

### 5-7. paddle.geometric.send_[u_recv ue_recv uv]

在 https://github.com/PFCCLab/PaddleAPITest/pull/23 中直接通过测试，但是在其他环境中报错。经检查是 paddle.geometric.send_ue_recv 的 `src_index`、`dst_index` 参数位置与另外两个 API 不同，因此修改生成代码

## 其他修改

***config_analyzer.py***
- 强化了 check_arg 与 get_arg 函数，使其必须同时接受参数位置索引与名称，增强配置解析的鲁棒性（配置可能提供了关键字，也可能没有）

## 测试结果

ctc_loss dice_loss 相关配置的 paddle_only 测试全部通过，cross_entropy 相关配置部分通过

ctc_loss dice_loss 配置位于 `api_config_merged_not_support.txt` ，移入 `api_config_merged_7.txt` 中

cross_entropy 经测试，主要有三个问题：类型不匹配、内存分配不足。经检查，类型不匹配的问题可以通过添加 `--test_amp=True` 解决，因此将相关配置移入 `api_config_merged_amp.txt` ；内存错误写入 fresh_report 中，与成功通过测试的配置一齐移入 `api_config_merged_tensor_init.txt`

```
test begin: paddle.nn.functional.cross_entropy(Tensor([3, 2, 2, 5],"float64"), Tensor([3, 2, 2, 5],"float64"), soft_label=True, label_smoothing=0.18833946207874966, axis=-1, weight=Tensor([5],"float64"), reduction="mean", )
W0410 02:23:01.192503 15404 backward.cc:437] While running Node (MatmulGradNode) raises an EnforceNotMet exception
[paddle error] paddle.nn.functional.cross_entropy(Tensor([3, 2, 2, 5],"float64"), Tensor([3, 2, 2, 5],"float64"), soft_label=True, label_smoothing=0.18833946207874966, axis=-1, weight=Tensor([5],"float64"), reduction="mean", ) 
 (PreconditionNotMet) Tensor's dimension is out of bound.Tensor's dimension must be equal or less than the size of its memory.But received Tensor's dimension is 480, memory's size is 256.
  [Hint: Expected numel() * SizeOf(dtype()) <= memory_size(), but received numel() * SizeOf(dtype()):480 > memory_size():256.] (at ../paddle/phi/core/dense_tensor_impl.cc:48)
```

image

### CODE REVIEW

File: report/fresh_report/test_log_paddle.nn.functional.cross_entropy.txt [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+paddle.nn.functional.cross_entropy(Tensor([4, 3],"float64"), Tensor([4],"int64"), soft_label=True, label_smoothing=0.1858912794612338, axis=-1, weight=Tensor([3],"float64"), reduction="mean", )
+paddle.nn.functional.cross_entropy(Tensor([4, 3],"float64"), Tensor([4],"int64"), soft_label=True, label_smoothing=0.5436483450993637, axis=-1, weight=None, reduction="none", )
+
+2025-04-10 02:22:56.100180 test begin: paddle.nn.functional.cross_entropy(Tensor([16, 400],"float16"), Tensor([16, 400],"float32"), soft_label=True, )
+W0410 02:23:00.687601 15369 gpu_resources.cc:119] Please NOTE: device: 0, GPU Compute Capability: 6.1, Driver API Version: 12.0, Runtime API Version: 11.8
+W0410 02:23:00.688882 15369 gpu_resources.cc:164] device: 0, cuDNN Version: 8.8.
+[paddle error] paddle.nn.functional.cross_entropy(Tensor([16, 400],"float16"), Tensor([16, 400],"float32"), soft_label=True, ) 
+ (InvalidArgument) The Input(Label) should be with the same data type as Input(Logits).
+  [Hint: Expected dtype == phi::CppTypeToDataType<T>::Type(), but received dtype:10 != phi::CppTypeToDataType<T>::Type():15.] (at ../paddle/phi/kernels/gpu/cross_entropy_kernel.cu:1417)
```
Comment (wanghuancoder) [inline report/fresh_report/test_log_paddle.nn.functional.cross_entropy.txt:13]: 这个报错看起来这个配置应该是一个AMP的配置。你可以问问皇润，他支持了AMP。你用AMP测一下，如果可以通过，可以把这个配置放到AMP专门的配置文件中去。

Comment (cangtianhuang) [reply report/fresh_report/test_log_paddle.nn.functional.cross_entropy.txt:13]: 使用了 amp 后仍未通过测试，看上去似乎是 gpu 的问题：
```
(paddle) λ yq01-sys-hic-k8s-p40-qm-0019 ~/workspace/PaddleAPITest python engine.py --paddle_only=True --api_config_file="tester/api_config/api_config_merged_temp.txt" --test_amp=True
grep: warning: GREP_OPTIONS is deprecated; please use an alias or script
2025-04-10 03:46:18.977977 test begin: paddle.nn.functional.cross_entropy(Tensor([16, 400],"float16"), Tensor([16, 400],"float32"), soft_label=True, )

/root/miniconda3/envs/paddle/lib/python3.12/site-packages/paddle/amp/auto_cast.py:634: UserWarning: For float16, amp only support NVIDIA GPU with Compute Capability 7.0 or higher, current GPU is: Tesla P40, with Compute Capability: 6.1.
  warnings.warn(
W0410 03:46:23.768945  1371 gpu_resources.cc:119] Please NOTE: device: 0, GPU Compute Capability: 6.1, Driver API Version: 12.0, Runtime API Version: 11.8
W0410 03:46:23.770135  1371 gpu_resources.cc:164] device: 0, cuDNN Version: 8.8.
[paddle error] paddle.nn.functional.cross_entropy(Tensor([16, 400],"float16"), Tensor([16, 400],"float32"), soft_label=True, ) 
 (InvalidArgument) The Input(Label) should be with the same data type as Input(Logits).
  [Hint: Expected dtype == phi::CppTypeToDataType<T>::Type(), but received dtype:10 != phi::CppTypeToDataType<T>::Type():15.] (at ../paddle/phi/kernels/gpu/cross_entropy_kernel.cu:1417)

Comment (cangtianhuang) [reply report/fresh_report/test_log_paddle.nn.functional.cross_entropy.txt:13]: 可以暂时将这部分放在not support中

Comment (cangtianhuang) [reply report/fresh_report/test_log_paddle.nn.functional.cross_entropy.txt]: 换了一个更高版本的gpu，成功通过了测试

**Review by wanghuancoder — APPROVED** [review]
LGTM
